amqp/Go/amqpPut.go

```diff
@@ -0,0 +1,188 @@
+    Port             string `json:"PORT"`
+}
+
+type MQEndpoints struct {
+    Points []Env `json:"MQ_ENDPOINTS"`
+}
+
+var (
+    queueName   string
+    topicName   string
+    valid       bool
+    destination string
+    isQueue     bool
+)
+var EnvSettings Env
+var MQ_ENDPOINTS MQEndpoints
+
+func init() {
+    //Parsing the input
+    flag.StringVar(&queueName, "q", "DEMO", "Queue")
+    flag.StringVar(&topicName, "t", "TOP", "Topic")
+    flag.Parse()
+    if flag.NFlag() > 2 {
+        valid = false
+    } else if flag.NFlag() == 1 {
+        flag.Visit(func(f *flag.Flag) {
+            if f.Name == "q" || f.Name == "t" {
+                valid = true
+            }
+            if f.Name == "q"{
+                destination = queueName
+                isQueue = true
+            }else{
+                destination = topicName
+            }
+        })
+    } else {
+        valid = false
+    }
+
+    if valid{
+        // ***********Configuring MQ Credentials***********
+        // Read MQ configuration from env.json
+        //Check for file path in env variables else default to ../env.json
+        DEFAULT_ENV_FILE := "../../env.json"
+        filePath := os.Getenv("ENV_FILE")
+        if filePath != ""{
+            fmt.Println("ENV file is set to " + filePath)
+        } else{
+            fmt.Println("Using default ENV file: "+DEFAULT_ENV_FILE)
+            filePath = DEFAULT_ENV_FILE
+        }
+        content, err := ioutil.ReadFile(filePath)
+        if err != nil {
+            log.Fatal("Error when opening file: ", err)
+        }
+        // Unmarshall the data
+        err = json.Unmarshal(content, &MQ_ENDPOINTS)
+        if err != nil {
+            log.Fatal("Error during Unmarshal(): ", err)
+        }
+        if len(MQ_ENDPOINTS.Points) > 0 {
+            EnvSettings = MQ_ENDPOINTS.Points[0]
+        }
+    }
+}
+
+func main() {
+    if !valid || queueName=="" || topicName==""{
+        fmt.Println("Invalid arguments")
+        return
+    }
+
+    //Defining the AMQP URI
+    uri:= "amqp://"+EnvSettings.Username+":"+EnvSettings.Password+"@"+EnvSettings.Host+":"+EnvSettings.Port
+
+    // Set up connection parameters
+    conn, err := amqp.Dial(context.TODO(), uri, nil)
+    fmt.Println("Connected to host.")	
+    if err != nil {
+        fmt.Println("Error in creating connection.")
+        log.Fatal(err)
+    }
+
+    // Open a session
+    session, err := conn.NewSession(context.TODO(), nil)
+    fmt.Println("Session created.")
+    if err != nil {
+        fmt.Println("Error in creating Session")
+        log.Fatal(err)
+    }
+
+    //When the destination is a queue, capability must be set to "queue". If not, topic is selected by default.
+    SenderOptions := &amqp.SenderOptions{
+        TargetCapabilities: []string{"topic"},
+    }
+    if isQueue{
+        SenderOptions = &amqp.SenderOptions{
+            TargetCapabilities: []string{"queue"},
+        }
+    }
+
+    // Create a sender

```

Comment:
Break down the code into methods / functions to make each unit of functionality easer to read.